### PR TITLE
[DI][Routing] Allow invokable objects to be used as PHP-DSL loaders

### PR DESCRIPTION
**Pull request by aurimasniekis**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26583, #25630
| License       | MIT
| Doc PR        | none

Changed DI/Router PHPFileLoader to check is_object && is_callable instead of instance of Closure

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thanks :)
Would you mind adding a test case please?

**Comment by aurimasniekis** [comment]
For which case? Then tried to supply array callback? Because then array would be passed nothing will happen so I don't know what to test against.

**Comment by nicolas-grekas** [comment]
For the case when you pass an invokable object.

**Comment by aurimasniekis** [comment]
Oh right forgot about this simple case 😊

**Comment by aurimasniekis** [comment]
@nicolas-grekas I don't know about Routing test I did not want to introduce copied test case so I just extended existing test case there.

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/object.expected.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/object.expected.yml]: extra unneeded line

Comment (aurimasniekis) [reply src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/object.expected.yml]: Every expected file has it

https://github.com/aurimasniekis/symfony/blob/be54d774ea5fd4cf9496fb4b39fc81c9ddefc25f/src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/basic.expected.yml#L1

**Review by Tobion — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Pull request merged**